### ISSUE TITLE
Deleting readonly class members does not give type error

### ISSUE THREAD

**Issue by r00ger** [Microsoft/TypeScript#11730]
As @mhegazy [said](https://github.com/Microsoft/TypeScript/issues/6717#issuecomment-254639267) in #6717 deleting of readonly properties should follow the same rules as writing.

**TypeScript Version:**  2.0.3 

**Code**

```ts
class Foo {
    public readonly bar: Bar;
    constructor() {
        this.bar = createBar(...);
    }
    public destroy() {
        delete this.bar; // here
    }
}
```

**Expected behavior:**
`tsc` should give a compile-time error
**Actual behavior:** 
No error

**Comment by r00ger** [comment]
Associated with #11480

**Comment by RyanCavanaugh** [comment]
Any change we do in #11480 would apply equally here, so we can track the change at that issue. Thanks!
